Desktop/analysis/analysis.cpp

```diff
@@ -727,6 +732,26 @@ std::string Analysis::upgradeMsgsForOption(const std::string & name) const
 	return out.str();
 }
 
+std::vector<std::string> Analysis::getValuesFromRSource(const QString &sourceID) const
+{
+	const Json::Value& jsonValue = _RSources.count(fq(sourceID)) == 0 ? Json::nullValue : _RSources.at(fq(sourceID));
+	std::vector<std::string> result;
+
+	if (!jsonValue.isObject()) return result;
+
+	const Json::Value& dataValue = jsonValue["data"];
+	for (const Json::Value& rowValue : dataValue)
+		for (const Json::Value& colValue : rowValue)
+		{
+			if (colValue.isString())
```

Comment:
I really think you should concatenate the columns if they are in the same row. Otherwise it will get confusing.